cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceAdmin.java

```diff
@@ -71,4 +70,13 @@
    * @throws NamespaceCannotBeDeletedException if the deletion operation was unsuccessful
    */
   public void deleteNamespace(Id.Namespace namespaceId) throws NotFoundException, NamespaceCannotBeDeletedException;
+
+  /**
+   * Update namespace properties for a given namespace.
+   *
+   * @param namespaceId  the {@link Id.Namespace} of the namespace to be updated
+   * @param properties  Map of properties to be updated
+   * @throws NotFoundException if the specified namespace is not found
+   */
+  public void updateProperties(Id.Namespace namespaceId, Map<String, String> properties) throws NotFoundException;
```

Comment:
Maybe the create API could update name and description if the namespace exists, else create a new one?